Add Microsoft.Communication resource manager version 2020-08-20

memontic-ms opened the pull request:
Bringing over the swagger and README changes for GA.
Stable version 2020-08-20.

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  Now
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  As soon as possible before March
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

memontic-ms committed:
Add Microsoft.Communication resource manager version 2020-08-20

allenjzhang commented on specification/communication/resource-manager/Microsoft.Communication/stable/2020-08-20/CommunicationService.json:
```diff
@@ -0,0 +1,1118 @@
+            "description": "Parameters for the update operation",
+            "required": false,
+            "schema": {
+              "$ref": "#/definitions/TaggedResource"
```
TaggedResource? Not CommunicationServiceResource?

memontic-ms replied:
I'm not sure of the reason for this. Tagging in @bviglietta who might have an answer as to why.

bviglietta replied:
I'm not sure why it's TaggedResource, since it does seem like CommunicationServiceResource would make more sense here. Although CommunicationServiceResource doesn't actually have any mutable fields that you could patch, so it doesn't make a huge difference.

memontic-ms replied:
@allenjzhang Is this blocking our signoff? This part hasn't changes since our previous version.

allenjzhang replied:
If it hasn't changed, I won't block the PR. But please review with team whether it is correct. Thx

memontic-ms replied:
@allenjzhang This has now been updated to use CommunicationServiceResource instead of TaggedResource

allenjzhang commented:
/azp run

RamyasreeChakka commented on specification/communication/resource-manager/Microsoft.Communication/stable/2020-08-20/examples/checkNameAvailabilityAvailable.json:
```diff
@@ -0,0 +1,19 @@
+{
+  "parameters": {
+    "subscriptionId": "12345",
+    "api-version": "2020-80-20",
```
API version is incorrect (specified 80 instead of 08). Cross check other places too.

RamyasreeChakka commented on specification/communication/resource-manager/Microsoft.Communication/stable/2020-08-20/CommunicationService.json:
```diff
@@ -0,0 +1,1118 @@
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Communication Services Management API",
+    "version": "2020-08-20",
```
As you are already used 2020-08-20-preview, recommend to bump up the GA version to avoid confusion.
To easily identify the delta changes from previous preview api version, the recommendation is to copy the existing version as is into new directory structure for first commit and then push new changes in subsequent commits. Please raise a new PR as suggested for faster review.

usvoyager replied:
We asked the team earlier for the GA best practices and this is what was recommended, upgrade preview version to stable by removing the -preview suffix which actually tells the story that we worked on it before to reach stable version, and that made sense. Changing the api-version now will incur more changes in our backend service and arm manifest, and I would prefer not doing that so close to GA, the team is busy with stabilization for GA. Thanks for understanding

memontic-ms replied:
@RamyasreeChakka Is this issue blocking the ARM signoff of this review? We don't intend to change this. Let's discuss so that we can get this signed off.

memontic-ms committed:
Fix mistyped API version from 2020-80-20 to 2020-08-20

memontic-ms committed:
Fix linter errors and bring in more GA changes

- Remove 3 instances of ARM violation R4009 by including systemData field
- Change resource update to take a CommunicationServiceResource instead
  of a TaggedResource
- Remove custom definition for OperationList and use the common-types
  version instead
- Validate resource names

allenjzhang reviewed (approved):
Approved from SDK side. Please address ARM feedback and obtain ARMSignedOff. Thanks.

usvoyager reviewed (approved):
:shipit:

bviglietta reviewed (approved):
:shipit:

j5lim commented on specification/communication/resource-manager/Microsoft.Communication/stable/2020-08-20/CommunicationService.json:
```diff
@@ -0,0 +1,965 @@
+            }
+          },
+          "201": {
+            "description": "Created and an async operation is executing in background to make the new key take effect. The response contains new keys and an Azure-AsyncOperation header to query the async operation result.",
```
> "description": "Created and an async operation is executing in background to make t [](start = 11, length = 84)

Is this async? Then please follow the rpc guideline. [https://github.com/Azure/azure-resource-manager-rpc/blob/f39512268c61b0a970766515d535a756495bb56d/v1.0/Addendum.md#call-action-post-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/f39512268c61b0a970766515d535a756495bb56d/v1.0/Addendum.md#call-action-post-asynchronously)

usvoyager replied:
RegenerateKey is not async, will remove the 201 from responses

j5lim reviewed (changes requested):
.

j5lim reviewed (approved):
Approved from ARM side.
